bundles/org.openhab.binding.volumio/src/main/java/org/openhab/binding/volumio/internal/VolumioHandler.java

```diff
@@ -188,41 +199,70 @@ private void clearChannels() {
     }
 
     private void handleVolumeCommand(Command command) {
-        if (command instanceof PercentType) {
-            volumio.setVolume((PercentType) command);
+        VolumioService volumioLocal = volumio;
+
+        if (volumioLocal == null) {
+            logger.debug("Volumio service was not yet initialized");
+            if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "device is offline");
+            }
+            return;
+        }
+
+        if (command instanceof PercentType commandAsPercentType) {
+            volumioLocal.setVolume(commandAsPercentType);
         } else if (command instanceof RefreshType) {
-            volumio.getState();
+            volumioLocal.getState();
         } else {
             logger.error("Command is not handled");
         }
     }
 
     private void handleStopCommand(Command command) {
+        VolumioService volumioLocal = volumio;
+
+        if (volumioLocal == null) {
+            logger.debug("Volumio service was not yet initialized");
+            if (ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "device is offline");
+            }
+            return;
+        }
+
         if (command instanceof StringType) {
-            volumio.stop();
+            volumioLocal.stop();
             updateState(VolumioBindingConstants.CHANNEL_STOP, UnDefType.UNDEF);
         } else if (command.equals(RefreshType.REFRESH)) {
             updateState(VolumioBindingConstants.CHANNEL_STOP, UnDefType.UNDEF);
         }
     }
 
     private void handlePlaybackCommands(Command command) {
+        VolumioService volumioLocal = volumio;
+
+        if (volumioLocal == null) {
+            logger.debug("Volumio service was not yet initialized");
```

Comment:
Please remove logger and improve `ThingStatusDetail`

Reply:
solved